compiler/luci/pass/src/SubstitutePadV2ToPadPass.cpp

```diff
@@ -83,6 +83,94 @@
 namespace
 {
 
+struct Paddings
+{
+  struct Pad
+  {
+    int32_t front;
+    int32_t end;
+  };
+  /**
+   * @brief Store paddings position information.
+   * @details _padding_pos[k] stores Pad object at axis k
+   */
+  std::vector<Pad> _padding_pos;
+
+  Paddings(luci::CircleConst *paddings)
+  {
+    assert(paddings->dtype() == loco::DataType::S32);
+    assert(paddings->rank() == 2);
+    assert(paddings->dim(1).value() == 2);
+    assert(paddings->size<loco::DataType::S32>() == paddings->rank() * 4);
+
+    for (uint32_t i = 0; i < paddings->dim(0).value(); i++)
+    {
+      Pad pad{.front = paddings->at<loco::DataType::S32>(i * 2),
+              .end = paddings->at<loco::DataType::S32>(i * 2 + 1)};
+      _padding_pos.emplace_back(pad);
+    }
+  }
+
+  /**
+   * @brief Check if this padding area is covered by filter
+   *
+   * @note This is to check condition C4).
+   *       _padding_pos should store values according to NHWC.
+   */
+  bool smaller_than(int32_t filter_h, int32_t filter_w)
+  {
+    assert(_padding_pos.size() == 4);
```

Comment:
If there can be `_padding_pos.size() != 4` for any model, 
it would be better to `return false` if not met the condition
- debug build will stop, which will break the testing/development
- release will just go and may raise access violation.

if this is to break on purpose to check while developing, it is OK.

Reply:
That is a very good recommendation. I fixed code.